raycicmd/bk_pipeline.go

```diff
@@ -56,14 +56,38 @@ type stepDockerPluginConfig struct {
 	mountBuildkiteAgent bool
 }
 
-func makeRayDockerPlugin(
-	image string, config *stepDockerPluginConfig,
-) map[string]any {
+func dockerPluginEnvList(config *stepDockerPluginConfig) []string {
+	envs := append([]string(nil), buildkiteEnvs...)
+	if len(config.extraEnvs) > 0 {
+		envs = append(envs, config.extraEnvs...)
+	}
+	return envs
+}
+
+const windowsBuildEnvImage = "rayproject/buildenv:windows"
+
+func makeRayWindowsDockerPlugin(config *stepDockerPluginConfig) map[string]any {
 	envs := append([]string(nil), buildkiteEnvs...)
 	if len(config.extraEnvs) > 0 {
 		envs = append(envs, config.extraEnvs...)
 	}
 
+	m := map[string]any{
+		"image":          windowsBuildEnvImage,
+		"shell":          []string{"bash", "-c"},
+		"shm-size":       "2.5gb",
+		"mount-checkout": true,
```

Comment:
does this mount the checkout to `/ray` workdir?; also can we amount the docker.sock as well for consistency? the artifact-mount can probably be done at a later time.

Reply:
it checks out at a temp dir, and mount it into `c:\workdir` inside the image.

the docker sock probably does not work.. the socket is a "unix domain socket", and my educated guess is that it won't work on windows. google search says it needs to mount in the pipe.

I am trying to keep the status quo of how windows run on civ1 for now, and get it migrated first, and then we can iterate from there.